frontend/src/components/maps/leaflet/PointClusterer.tsx

```diff
@@ -89,47 +83,145 @@ export const PointClusterer: React.FC<PointClustererProps> = ({
       map.off('clear', spiderfier.unspiderfy, spiderfier);
       spiderfierRef.current = undefined;
     };
-  };
-  useEffect(componentDidMount, [map, onMarkerClick, supercluster]);
+  }, [map, onMarkerClick, supercluster]);
+
+  useEffect(componentDidMount, [componentDidMount]);
 
   // on-click handler
-  const zoomOrSpiderfy = (cluster: ICluster) => {
-    if (!supercluster || !spiderfierRef.current || !cluster) {
-      return;
-    }
-    const { cluster_id } = cluster.properties;
-    const expansionZoom = Math.min(
-      supercluster.getClusterExpansionZoom(cluster_id as number),
-      maxZoom as number,
-    );
-    // already at maxZoom, need to spiderfy child markers
-    if (expansionZoom === maxZoom && spiderfyOnMaxZoom) {
-      spiderfierRef.current.spiderfy(cluster);
-    } else if (zoomToBoundsOnClick) {
-      zoomToCluster(cluster, expansionZoom, map);
-    }
-  };
-
-  const onLayerClick = (e: LeafletMouseEvent) => {
-    // the point may be either a cluster or a single map pin
-    const clusterOrPin = (e?.propagatedFrom as Marker)?.feature as ICluster;
-    const isCluster = clusterOrPin?.properties?.cluster;
-
-    // the user clicked on a cluster
-    if (!!isCluster) {
-      zoomOrSpiderfy(clusterOrPin);
-    } else {
-      onMarkerClick?.(clusterOrPin as PointFeature); // single pin
-    }
-  };
+  const zoomOrSpiderfy = useCallback(
+    (cluster: ICluster) => {
+      if (!supercluster || !spiderfierRef.current || !cluster) {
+        return;
+      }
+      const { cluster_id } = cluster.properties;
+      const expansionZoom = Math.min(
+        supercluster.getClusterExpansionZoom(cluster_id as number),
+        maxZoom as number,
+      );
+      // already at maxZoom, need to spiderfy child markers
+      if (expansionZoom === maxZoom && spiderfyOnMaxZoom) {
+        const res = spiderfierRef.current.spiderfy(cluster);
+        setSpider(res);
+      } else if (zoomToBoundsOnClick) {
+        zoomToCluster(cluster, expansionZoom, map);
+      }
+    },
+    [spiderfierRef, map, maxZoom, spiderfyOnMaxZoom, supercluster, zoomToBoundsOnClick],
+  );
 
   return (
-    <GeoJsonLayer
-      ref={ref}
-      data={clusters as any}
-      pointToLayer={pointToLayer}
-      onclick={onLayerClick}
-    ></GeoJsonLayer>
+    <>
```

Comment:
Why was this change necessary?

Reply:
This is a pretty big render block with a lot of different categories of functionality. Any way we could split this up? Even just creating functions for these for readability would be fine.